File: src/Tree/BinaryTreeInorderTraversal/BinaryTreeInorderTraversal_Recursive.java
class Solution {
    List<Integer> res = new LinkedList<>();
    
    public List<Integer> inorderTraversal(TreeNode root) {
        dfs(root);
        return res;
    }
    private void dfs(TreeNode root){
        //base case
        if(root == null) return;
        
        //traverse the left side
        dfs(root.left);
        
        //take the current node val
        res.add(root.val);
        
        //traverse the right side
        dfs(root.right);
    }
}

File: src/Tree/BinaryTreeInorderTraversal/BinaryTreeInorderTraversal_Recursive.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        res = []
        
        def dfs(root: Optional[TreeNode]):
            if root is None:
                return
            
            dfs(root.left)
            
            res.append(root.val)
            
            dfs(root.right)
            
        dfs(root)
        
        return res
